src/server-app/editor-server.html

```diff
@@ -0,0 +1,159 @@
+<!--
+/**
+ * Copyright 2017, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<link rel="import" href="../../bower_components/polymer/polymer-element.html">
+
+<dom-module id="editor-server">
```

Comment:
I'm confused about the structure of how all this fits together. Is there a design doc/explainer of how this all fits together?

cc @e111077

Reply:
1. Developer creates an iframe with src pointing to this server
2. This server's CSP checks if it is an okay domain to embed the iframe
3. Server serves editor-server which communicates via messages through the iframe on what code to render
4. User specifies code to render
5. editor-server receives this message and creates a new iframe and inserts content
6. the new iframe may have html imports which it queries from server
7. server communicates via websockets with this element to determine whether or not the file is user-generated (not in bower_components / node_modules)
8. server serves the content and we don't have any 404s and everything is super duper happy 😀